x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/description_step/index.tsx

```diff
@@ -128,55 +127,34 @@ const getDescriptionItem = (
       indexPatterns,
     });
   } else if (field === 'threat') {
-    const threat: IMitreEnterpriseAttack[] = get(field, value).filter(
+    const threat: IMitreEnterpriseAttack[] = get(field, data).filter(
       (singleThreat: IMitreEnterpriseAttack) => singleThreat.tactic.name !== 'none'
     );
     return buildThreatDescription({ label, threat });
-  } else if (field === 'description') {
-    return [
-      {
-        title: label,
-        description: get(field, value),
-      },
-    ];
   } else if (field === 'references') {
-    const urls: string[] = get(field, value);
+    const urls: string[] = get(field, data);
     return buildUrlsDescription(label, urls);
   } else if (field === 'falsePositives') {
-    const values: string[] = get(field, value);
+    const values: string[] = get(field, data);
     return buildUnorderedListArrayDescription(label, field, values);
-  } else if (Array.isArray(get(field, value))) {
-    const values: string[] = get(field, value);
+  } else if (Array.isArray(get(field, data))) {
+    const values: string[] = get(field, data);
     return buildStringArrayDescription(label, field, values);
   } else if (field === 'severity') {
```

Comment:
Woa...This is very abnormal code flow you are refactoring here.

```ts
  } else if (field === 'falsePositives') {
    const values: string[] = get(field, data);
    return buildUnorderedListArrayDescription(label, field, values);
  } else if (Array.isArray(get(field, data))) {
    const values: string[] = get(field, data);
    return buildStringArrayDescription(label, field, values);
  } else if (field === 'severity') {
```

It checks if it is an array rather than a field value part way down but not the the field switch? This is going to make future changes brittle and error prone depending on where we add the field value.

I would ask you do us all a favor and add what the field value is through a call of:

```ts
} else if (field === 'whatever_string_value_this_is') {
```

That might not even be getting touched? It is really hard to understand code the more I look at it.

Reply:
Deferred to https://github.com/elastic/kibana/issues/60567